drivers/bluetooth/hci/hci_nxp.c

```diff
@@ -0,0 +1,421 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* -------------------------------------------------------------------------- */
+/*                                  Includes                                  */
+/* -------------------------------------------------------------------------- */
+
+#include <zephyr/init.h>
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/bluetooth/hci_types.h>
+#include <soc.h>
+
+#include <fwk_platform_ble.h>
+#include <fwk_platform.h>
+
+/* -------------------------------------------------------------------------- */
+/*                                  Definitions                               */
+/* -------------------------------------------------------------------------- */
+
+#define DT_DRV_COMPAT nxp_hci_ble
+
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+LOG_MODULE_REGISTER(bt_driver);
+
+#define HCI_IRQ_N        DT_INST_IRQ_BY_NAME(0, hci_int, irq)
+#define HCI_IRQ_P        DT_INST_IRQ_BY_NAME(0, hci_int, priority)
+#define HCI_WAKEUP_IRQ_N DT_INST_IRQ_BY_NAME(0, wakeup_int, irq)
+#define HCI_WAKEUP_IRQ_P DT_INST_IRQ_BY_NAME(0, wakeup_int, priority)
+
+#define HCI_CMD 0x01
+#define HCI_ACL 0x02
+#define HCI_EVT 0x04
```

Comment:
Seems like these could use the `BT_HCI_H4_*` defines from `<zehyr/bluetooth/hci_types.h>`